### ISSUE TITLE
-Wc has a limit of 128 characters

### ISSUE THREAD

**Issue by Rampoina** [freebasic/fbc#298]
## Expected Behavior
All of the flags specified by -Wc are correctly passed to gcc

## Current Behavior
The gcc flags specified by -Wc are cut after 127 characters

## Steps to Reproduce
 ```
fbc -gen gcc -Wc -option-option-option-option-option-option-option-option-option-option-option-option-option-option-option-option-option,-somethingelse -c test.bas
```
```
gcc: error: unrecognized command-line option ‘-somet’
```
the last option gets passed as -somet

**Comment by rversteegen** [comment]
Yes, this is pain, which I've hit many times, such as right now. It's not just -Wc, similar arguments also have arbitrary limits. (The fact that you can't pass -Wc/etc multiple times to fbc, as to can to gcc, is also very annoying.)
I suppose the limit originates from back when FB didn't have dynamic length strings, and I'm surprised all these fixed length strings in the compiler are still kept around. Is there a reason for that?

**Comment by jayrm** [comment]
> The fact that you can't pass -Wc/etc multiple times to fbc, as to can to gcc, is also very annoying

It's possible to pass `-Wc` multiple times to fbc.  It need not be grouped all in one.
This issue was reported in #137 Final -Wl flag takes precedence over previous
Fix applied in PR #138 Pass down all options from all -Wa, -Wc, and -Wl flags

PR #302 now using variable length string internally to remove the fixed string limit
